AstroAdapt.GraphQL/Program.cs

```diff
@@ -16,14 +17,17 @@
 builder.Services.AddSingleton<SolutionProcessingService>();
 
 builder.Services.AddGraphQLServer()
-    .AddProjections()
-    .AddFiltering()
-    .AddSorting()
     .AddQueryType<Query>()
-    .AddType<SolutionInputType>()
     .AddMutationType<Mutation>()
-    .AddTypeExtension<ComponentExtensions>()
     .AddSubscriptionType<Subscription>()
+    .AddAstroTypes()
+    .AddProjections()
+    .AddFiltering()
+    .AddSorting()
+    .RegisterService<IAstroApp>()
+    .RegisterService<ITopicEventReceiver>()
+    .RegisterService<ITopicEventSender>()
+    .RegisterDbContext<AstroContext>(DbContextKind.Pooled)
```

Comment:
Here we are registering our common services and our DBContext. Since we registered the DBContext as pooled we no longer have to deal with the `IDbContextFactory` in our resolver, since the execution engine will take care of renting and returning the DBContext.